tuxemon/animation.py

```diff
@@ -16,54 +21,68 @@
 ANIMATION_FINISHED = 3
 
 
-def is_number(value):
-    """Test if an object is a number.
-    :param value: some object
-    :returns: True
-    :raises: ValueError
+def check_number(value: Any) -> float:
+    """
+    Test if an object is a number.
+
+    Raises ``ValueError`` when ``value`` is not a number.
+
+    Parameters:
+        value: Some object.
+
     """
     try:
         float(value)
+        return value
     except (ValueError, TypeError):
         raise ValueError
 
-    return True
 
+def remove_animations_of(target: object, group: pygame.sprite.Group) -> None:
+    """
+    Find animations that target objects and remove those animations.
 
-def remove_animations_of(target, group):
-    """Find animations that target objects and remove those animations
+    Parameters:
+        target: Object whose animations should be removed.
+        group: Pygame group where to remove the animations.
 
-    :param target: any
-    :param group: pygame.sprite.Group
-    :returns: None
     """
     animations = [ani for ani in group.sprites() if isinstance(ani, Animation)]
     to_remove = [ani for ani in animations if target in [i[0] for i in ani.targets]]
     group.remove(*to_remove)
 
 
-class AnimBase(pygame.sprite.Sprite):
-    _valid_schedules = []
+class TaskBase(pygame.sprite.Sprite):
+    _valid_schedules: Sequence[str] = []
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
-        self._callbacks = defaultdict(list)
-
-    def schedule(self, func, when=None):
-        """Schedule a callback during operation of Task or Animation
+        self._callbacks: DefaultDict[
+            str,
+            List[ScheduledFunction],
+        ] = defaultdict(list)
+
+    def schedule(
+        self,
+        func: ScheduledFunction,
+        when: Optional[str] = None,
```

Comment:
Im not sure how this got to be a string, but probably should be `Union[int, float]` or something like that.  For tasks, the time unit is technically unitless; determining to execute or not is a comparison between two objects, but i think for clarity, an int or float is more likely to be used.  Again, nothing is enforced here, because the class doesn't matter.... `Any` would probably be most accurate, even if confusing.

Reply:
`when`  can be `"on interval", "on finish", "on abort"`.

Reply:
Maybe I should say "event" instead of "time" but I also see the confusion potential with that.

Reply:
maybe just change it to `Any` since that is the most correct.  any confusing part of it could be clarified in a comment if needed.

Reply:
But it cannot be a number. Just a string.

Reply:
it can be any hashable object.  the mechanism relies on a dictionary.  
```
callbacks = self._callbacks[when]
```
Is `hashable` a type?

Reply:
numbers are hashable and would work with this object
```
KeyError: 0
>>> d[0] = None
>>> d[1.0] = None
>>> d.keys()
[0, 1.0]
```

Reply:
Yes, it is a type.

However, `when` is a member of `self._valid_schedules` and the elements of that are always strings.

Reply:
honestly, im not really sure what to do.  `str` is fine.  lets just leave it, sorry for the confusion.